tools/lsp/server_loop.rs

```diff
@@ -420,11 +420,11 @@ pub async fn reload_document(
     Ok(())
 }
 
-fn get_document_and_offset(
-    document_cache: &mut DocumentCache,
-    text_document: lsp_types::TextDocumentIdentifier,
-    pos: Position,
-) -> Option<(&i_slint_compiler::object_tree::Document, u32)> {
+fn get_document_and_offset<'a>(
+    document_cache: &'a mut DocumentCache,
+    text_document: &'a lsp_types::TextDocumentIdentifier,
+    pos: &'a Position,
+) -> Option<(&'a i_slint_compiler::object_tree::Document, u32)> {
```

Comment:
The use of `text_document.uri` in this function suggests that `&Uri` would be the best choice IMO.